syft/generic/frameworks/hook/pointers.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC
+from functools import wraps
+
+import syft
+
+from syft.exceptions import TensorsNotCollocatedException
+from syft.generic.frameworks.hook import hook_args
+from syft.generic.pointers.multi_pointer import MultiPointerTensor
+from syft.generic.pointers.object_pointer import ObjectPointer
+from syft.generic.pointers.pointer_tensor import PointerTensor
+
+
+class PointerHook(ABC):
```

Comment:
I know your PR is only about moving some things around but do you know why we have this hooking on Pointers?
I thought this hooks where here to replace standard (non PySyft) behavior. But Pointers are part of PySyft, right?

Reply:
I’m not entirely sure, TBH. Maybe because they imitate native tensors?

Reply:
A lot of hooking is just a place to put for loops which add functionality (monkeypatching/codegen more than hooking). This is perhaps the biggest culprit.